File: Backend/src/modules/live_data_processor/domain_value/heal.rs
use crate::modules::live_data_processor::domain_value::{HitMask, Mitigation, Unit};

#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
pub struct Heal {
    pub total: u32,
    pub effective: u32,
    pub mitigation: Vec<Mitigation>,
    pub target: Unit,
    pub hit_mask: HitMask,
}

File: Webclient/src/app/module/viewer/domain_value/heal.ts
import {Mitigation} from "./mitigation";
import {Unit} from "./unit";
import {HitType} from "./hit_type";

export interface Heal {
    spell_cause_id: number;
    heal: {
        total: number;
        effective: number;
        mitigation: Array<Mitigation>;
        target: Unit;
        hit_mask: Array<HitType>;
    };
}
